Tensile/SolutionStructs.py

```diff
@@ -857,10 +858,108 @@ def __init__(self, config):
           if anchorDim not in self.state["IndexAssignments%s"%tc]:
               printExit("SetConstStride%s=%s anchorDim=%u is not in IndexAssignments%s"%(tc, sc, anchorDim, tc))
 
+  ################################################################################
+  # Assign and check the 3-datatypes EXPLICTLY for better maintanence:
+  # Target: to avoid some hack-code/workaround/ambiguous code
+  #     - Such as fixing Cexternal to DestDataType instead of DataType,
+  #       but now it works well since they are the same
+  #     - Another exmple is the "ComputeDataType" is barely referred in generator
+  #       We use lots of if (DataType==... and HPA==...) which might be tedious
+  #       It's better to make them explict
+
+  # Align the supported GEMM type with rocBLAS: [a/b/c/d/alpha/beta]
+  #   (rocblas/library/include/internal/rocblas_functions.h)
+  # Non_Ex
+  #   - CGEMM:  [C/C/ C/C/ C/C]
+  #   - ZGEMM:  [Z/Z/ Z/Z/ Z/Z]
+  #   - DGEMM:  [D/D/ D/D/ D/D]
+  #   - SGEMM:  [S/S/ S/S/ S/S]
+  #   - HGEMM:  [H/H/ H/H/ H/H] (HPA=F)
+  # _Ex:
+  #   - HGEMM:  [H/H/ H/H/ H/H] (HPA=T), SEE MORE EXPLANATION BELOW
+  #   - HGEMM:  [H/H/ H/H/ S/S] (HPA=T), Somehow we hadn't used this version
+  #   - BFGEMM: [B/B/ B/B/ S/S] (HPA=T), the only supported one
+  #   - IGEMM:  [i8/i8/ I/I/ I/I ], tensile packs 4 i8 to i8x4 with some restrictions
+
+  # DataType        = Input data-type of A/B
+  # DestDataType    = Output data-type of C/D
+  # ComputeDataType = Input data-type of alpha/beta:
+  # Cinternal: basically should == ComputeDataType
+  #   - EXCEPT FOR (H/H/H/H/H/H)+HPA, Cinternal is F32, which is not intuitive
+
+  # But for HGEMM, ComputeDataType is trickier:
+  #   - When rocBLAS invokes with a/b=F16, c/d=F16, compute=F16,
+  #     rocBLAS directly find a well-mapped tensile-kernel
+  #     -> Tensile: [H/H/ H/H/ H/H] & HPA=FALSE is used
+  #        Input Alpha/Beta is Half
+  #
+  #   - When rocBLAS invokes with a/b=F16, c/d=F16, compute=F32,
+  #     rocBLAS turns on HPA, then casts to a compute-type=F16 ver of tensile-kernel
+  #     -> Tensile: [H/H/ H/H/ H/H] & HPA=TRUE is used
+  #        Input Alpha/Beta is still Half
+  #     Which seems not a intuitive mapping for alpha/beta
+  #
+  # Target: The better and consistent logic should be the same as BF16
```

Comment:
Yes, this PR will allow Tensile to tune and run this type (H/H/S) HPA. The newly added pre_checkin test includes this. 
The last thing we haven't done is to modify rocblas to let "-compute-type=f32" (H/H/S) mapped to real (H/H/S) kernel instead of (H/H/H) HPA.
But once we do this, we need to make sure all the logic yamls in rocblas are  updated to (HHS), kind of a very careful work.

Reply:
We [workaround](https://github.com/ROCmSoftwarePlatform/rocBLAS-internal/blob/ba93621cf50c139683d1adc4ba426afd7e183b74/library/src/tensile_host.cpp#L270) this issue in rocBLAS, but the workaround can be removed once Tensile is fixed.